src/stores/store.js

```diff
@@ -20,29 +20,33 @@ export default new Vuex.Store({
   state: function () {
     return {
       todos: [],
-      limit: 4,
+      limit: 8,
+      currentFilter: "all",
       isSearching: false,
+      allTodoLen: 0,
+      incTodoLen: 0,
+      comTodoLen: 0,
+      toasts: [],
     }
   },
 
   getters: {
     getTodos(state) {
-      return state.todos.slice(0, state.limit)
-    },
+      let option = state.currentFilter
 
-    getTodo: (state) => (id) => {
-      const index = state.todos.findIndex((todo) => todo.id === parseInt(id))
+      let completedTodo = state.todos.filter((todo) => todo.done)
+      let inCompletedTodo = state.todos.filter((todo) => !todo.done)
 
-      if (index === -1) {
-        return {
-          status: "not-found",
-          todo: null,
-        }
+      state.allTodoLen = state.todos.length
+      state.comTodoLen = completedTodo.length
+      state.incTodoLen = inCompletedTodo.length
+
+      if (option === "all") {
+        return state.todos.slice(0, state.limit)
+      } else if (option === "com") {
+        return completedTodo.slice(0, state.limit)
```

Comment:
Instead of filtering in state, maintain the filterValue (all, complete, incomplete) in state and do the filtering in the component.